### DISCUSSION TITLE
Restart all the application context

### DISCUSSION THREAD

**Discussion by shaymolho** [airsdk/Adobe-Runtime-Support#1937] [Q&A]
There are some scenarios which we like to simulate a new login to the application
I saw there is an ANE https://docs.airnativeextensions.com/docs/restartapp restart app but it's only available in android so it's not good enough
is there a way in flash to clear the root application context so all static fields are cleared and start from scratch again?

**Comment by shaymolho** [comment]
@ajwfrost any idea?

**Reply by ajwfrost** [reply]
Interesting idea .. but I don't think this is currently a possibility, if you mean to completely clear the whole application and start again. The way the ActionScript Virtual Machine works, definitions are loaded in when the SWF is parsed, and you can't unload these (which means things like static variables will really be global/for the lifetime of the VM). So the only way for this to work would be if the VM is completely removed and restored.

The only thing I'm wondering is whether we could pick up the application exit when everything is being torn down, and then just go back into the application again as if from the first time. It depends a little on how each platform implementation is done though, and this would be a complete reset i.e. all windows and UI will be removed and then re-started. Of course, static variables in the C++ runtime code would still retain their earlier values so there are likely to be a number of challenges with this approach anyway....

It may just be simpler/safer to build a 'reset' into the SWF application itself?

thanks
